crates/nu-utils/standard_library/dirs.nu

```diff
@@ -0,0 +1,80 @@
+# Maintain a list of working directories and navigates them
+
+# the directory stack
+export-env {
+    let-env DIRS_POSITION = 0
+    let-env DIRS_LIST = [($env.PWD | path expand)]
+}
+
+# Add one or more directories to the list.
+# PWD becomes first of the newly added directories.
+export def-env "add" [
+    ...paths: string    # directory or directories to add to working list
+    ] {
+        mut abspaths = []
+        for p in $paths {
+            if ($p | path type) != 'dir' {
```

Comment:
you should `path expand` here as well otherwise
```bash
>_ dirs add ~/downloads/
Error:
  × not a directory
   ╭─[entry #15:1:1]
 1 │ dirs add ~/downloads/
   ·          ──────┬─────
   ·                ╰── not a directory
   ╰────
```

Reply:
will do.

Reply:
You're right, I've added the `path expand`.

But not for the right reason!  You should get the above error based in your example, because of case sensitive file system (I assume you have).
```nushell
〉use ~/src/rust/nushell/crates/nu-utils/standard_library/std.nu
---------------------------------------------------- /home/bobhy ----------------------------------------------------
〉std dirs add ~/downloads
Error: 
  × not a directory
   ╭─[entry #3:1:1]
 1 │ std dirs add ~/downloads
   ·              ─────┬─────
   ·                   ╰── not a directory
   ╰────

---------------------------------------------------- /home/bobhy ----------------------------------------------------
〉std dirs add ~/Downloads
----------------------------------------------- /home/bobhy/Downloads -----------------------------------------------
〉std dirs show
╭───┬────────┬───────────────────────╮
│ # │ active │         path          │
├───┼────────┼───────────────────────┤
│ 0 │ false  │ /home/bobhy           │
│ 1 │ true   │ /home/bobhy/Downloads │
╰───┴────────┴───────────────────────╯
```

Reply:
> You're right, I've added the `path expand`.
>
thanks @bobhy